task_processing/plugins/persistence/file_persistence.py

```diff
@@ -0,0 +1,29 @@
+import json
+
+from pyrsistent import v
+
+from task_processing.interfaces.event import Event
+from task_processing.interfaces.event import json_deserializer
+from task_processing.interfaces.event import json_serializer
+from task_processing.interfaces.persistence.reader import Reader
+from task_processing.interfaces.persistence.writer import Writer
+
+
+class FilePersistence(Reader, Writer):
```

Comment:
I agree, I'd just have like a Persistence interface that has read (read_status) and write (write_event)